File: sdk/ruby/lib/telecom_sdk/graphql_api.rb
module TelecomSDK
  # API for GraphQL queries
  class GraphQLAPI
    def initialize(client)
      @client = client
    end

    def execute(query, variables = nil)
      request = { query: query }
      request[:variables] = variables if variables
      @client.post("/graphql", request)
    end
  end
end

File: sdk/elixir/lib/telecom_sdk/graphql_api.ex
defmodule TelecomSDK.GraphQLAPI do
  @moduledoc """
  API for GraphQL queries
  """

  defstruct [:client]

  def new(client) do
    %__MODULE__{client: client}
  end

  def execute(api, query, variables \\ nil) do
    request = %{query: query}
    request = if variables, do: Map.put(request, :variables, variables), else: request
    TelecomSDK.HTTPClient.post(api.client, "/graphql", request)
  end
end
